graphql_compiler/query_pagination/__init__.py

```diff
@@ -79,10 +79,10 @@ def paginate_query_ast(
 
     # See if we can and should split the query
     num_pages = 1
-    if query_analysis.classes_with_missing_counts:
+    if query_analysis._classes_with_missing_counts:
         advisories += tuple(
             MissingClassCount(class_name)
-            for class_name in query_analysis.classes_with_missing_counts
+            for class_name in query_analysis._classes_with_missing_counts
```

Comment:
This code block can be a helper method called `get_desired_number_of_pages(desired_page_size: int, ...)`, and that method can be part of the `QueryPlanningAnalysis` class, and can freely access the private field `_classes_with_missing_counts`.